.gitignore

```diff
@@ -1 +1,32 @@
-.vscode/
\ No newline at end of file
+# Prerequisites
+*.d
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+*.dll
+
+# Compiled Static libraries
```

Comment:
we don't use any of these why are we adding them in?

Reply:
Like I said in an earlier response, I was trying to be pro-active and I learned a bit about file extensions.  Feel free to take some or all of these out.  It really doesn't matter to me.